### PR TITLE
Port xtensa arch

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
-libmetal: add xtensa support
-libmetal: add __sync_synchronize support

### CODE REVIEW

File: lib/processor/xtensa/atomic.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2015, Xilinx Inc. and Contributors. All rights reserved.
```
Comment (arnopo) [inline lib/processor/xtensa/atomic.h:2]: please fix copyright in files

File: lib/processor/xtensa/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+collect (PROJECT_LIB_HEADERS atomic.h)
+collect (PROJECT_LIB_HEADERS cpu.h)
+
+# vim: expandtab:ts=2:sw=2:smartindent
```
Comment (arnopo) [inline lib/processor/xtensa/CMakeLists.txt:4]: useless line to clean

File: lib/processor/xtensa/cpu.h [hunk 3]
```diff
@@ -13,5 +13,6 @@
 #define __METAL_XTENSA_CPU__H__
 
 #define metal_cpu_yield()
+#define __sync_synchronize()
```
Comment (arnopo) [inline lib/processor/xtensa/cpu.h:16]: any reason to not squash this commit with the previous one?

**Review by edmooring — CHANGES_REQUESTED** [review]
Please address Arnaud's feedback. Otherwise, it looks good to me.

File: lib/processor/xtensa/CMakeLists.txt (hunk 2 above)
Comment (xiaoxiang781216) [reply lib/processor/xtensa/CMakeLists.txt]: Done.

File: lib/processor/xtensa/atomic.h (hunk 1 above)
Comment (xiaoxiang781216) [reply lib/processor/xtensa/atomic.h]: Done.

File: lib/processor/xtensa/cpu.h (hunk 3 above)
Comment (xiaoxiang781216) [reply lib/processor/xtensa/cpu.h:16]: Done.

**Review by edmooring — APPROVED** [review]
Looks good to go.

**Comment by lenghonglin** [comment]
Hello, @xiaoxiang781216  This pr will not build passed for me with zephyr , the error message is missing sys. c and sys.h . and i create new dir in this path `system/zephyr/xtensa` and port sys.c and sys.h , the problem is be sloved.

image

**Comment by xiaoxiang781216** [comment]
> Hello, @xiaoxiang781216 This pr will not build passed for me with zephyr , the error message is missing sys. c and sys.h . and i create new dir in this path `system/zephyr/xtensa` and port sys.c and sys.h , the problem is be sloved.
> 
> image

I use NuttX porting, not Zephyr porting. it look like that sys.[h|c] is a Zephyr specific requirement. Could you create a PR fix this issue?

**Comment by dbaluta** [comment]
@xiaoxiang781216 @lenghonglin I also met this issue. Will try to send a PR to fix it!

**Comment by lenghonglin** [comment]
Awwsome~~

**Pull request merged**